tests/php/src/DevTools/CallbackReflectionTest.php

```diff
@@ -100,14 +100,6 @@ public function data_get_source() {
 				'widgets/class-wp-widget-text.php',
 				ReflectionMethod::class,
 			],
-			'core_widget_display_callback_string' => [
-				'WP_Widget_Text::display_callback',
```

Comment:
@westonruter This was also a non-static method and the test would fail on PHP 8. Would this need to be replaced with another method or would the overall test data here be sufficient?

Reply:
The callback isn't being called this way in core anyway, so this should be replaced with:

```php
[ new WP_Widget_Text(), 'display_callback' ]
```

Reply:
That would make it the same test as the one above (`core_widget_display_callback_array`), no?

Reply:
I suppose yes, but why is `core_widget_display_callback_array` working currently since `WP_Widget::display_callback` is not static?

Reply:
I noticed there's a condition handling if this specific method was the callback:

https://github.com/ampproject/amp-wp/blob/9f64f1137573d0263694bc7fa3446f764da35d7f/src/DevTools/CallbackReflection.php#L115-L124

Reply:
But in the `core_widget_display_callback_array` test data, the supplied callback is `[ 'WP_Widget_Text', 'display_callback' ]`. This will then get passed into:

```php
$reflection = new ReflectionMethod( $callback[0], $callback[1] );
```

Since `display_callback` is not static, I thought that `ReflectionMethod` would have been an error here?

Reply:
It shouldn't, since it's not checking if the method is callable when instantiating `ReflectionMethod`. It should only be reporting  information about the method.

Reply:
Oh, I see. Thanks. So then yes, let's remove the `core_widget_display_callback_string` data (as you've already done), and then change `core_widget_display_callback_array` to use `[ new WP_Widget_Text(), 'display_callback' ]` instead of `[ 'WP_Widget_Text', 'display_callback' ]` to reflect how it is actually used in WP.

Reply:
Done in https://github.com/ampproject/amp-wp/pull/5475/commits/db61b4cfec1f63f9ef249defe7869956b7cc36f0.